src/libpmem/meson.build

```
@@ -0,0 +1,33 @@
+sources = files(
+    'libpmem.c',
+    '../libpmem2/memops_generic.c',
+    'pmem.c',
+    'pmem_posix.c',
+)
+
+libpmem = library(
+    'pmem',
+    sources,
+    c_args: [
+        '-Wl,--version-script=@0@'.format(meson.current_source_dir() / 'libpmem.link.in')
+    ],
+    dependencies: [
+        threads_dep,
+        libndctl_dep,
+        libdaxctl_dep,
+        pmemcommon_dep,
+        pmemcore_dep,
+        pmem2arch_dep,
+    ],
+    include_directories: root_include,
+    soversion: '1',
+    version: '1.0.0',
+    install: true
+)
+
+libpmem_dep = declare_dependency(
+    link_with: libpmem,
+    include_directories: include_directories('.')
+)
+
+meson.override_dependency('libpmem', libpmem_dep)
```

```suggestion
if meson.version().version_compare('>=0.54.0')
    meson.override_dependency('libpmem', libpmem_dep)
endif
```

It is also possible to trivially support the version of meson available in Ubuntu 20.04's official repositories, since all of these `meson.override_dependency` invocations are functionality which is not actually needed to build the project, only for convenience when using it as a wrap in another project.

So gracefully degrading is a reasonable option.

---

Right. I will make your suggestion here.